lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+// $FlowFixMe: Flow doesn't know about useState: https://stackoverflow.com/questions/53105954/cannot-import-usestate-because-there-is-no-usestate-export-in-react-flow-with
+import React, {useState} from 'react'
+import {Button, Modal} from 'react-bootstrap'
+
+import * as tripActions from '../../actions/trip'
+import {getComponentMessages} from '../../../common/util/config'
+import HourMinuteInput from '../HourMinuteInput'
+import type {Trip} from '../../../types'
+
+type Props = {
+  addNewTrip: typeof tripActions.addNewTrip,
+  constructNewRow: (trip: ?Trip, tripSeriesStartTime: number) => ?Trip,
+  onClose: () => void,
+  show: boolean
+}
+
+const initialState = {endTime: 0, error: '', startTime: 0, headway: 0}
+
+const TripSeriesModal = (props: Props) => {
+  const [{endTime, error, headway, startTime}, setState] = useState(initialState)
```

Comment:
I don't like the new combined state. I think the previous 3 states was much cleaner.

Reply:
Ok try this on for size: https://github.com/ibi-group/datatools-ui/pull/892/commits/e29b27fff2a7d41b88a3dfb1982b07101a3d2544